src/main/cpp/src/genomicsdb/variant_field_handler.cc

```diff
@@ -397,6 +397,102 @@ void  VariantOperations::remap_data_based_on_genotype(const std::vector<DataType
   }
 }
 
+//genotype corresponding to min value PL field
+
+template<class DataType>
+void GenotypeForMinValueTracker<DataType>::track_minimum(const std::vector<DataType>& input_data,
+    std::vector<int>& allele_idx_vec_for_current_gt_combination)
+{
+  auto gt_idx = VariantOperations::get_genotype_index(allele_idx_vec_for_current_gt_combination, false);
+  if(gt_idx < input_data.size() && is_bcf_valid_value<DataType>(input_data[gt_idx])
+      && input_data[gt_idx] < m_current_min_value )
+  {
+    m_current_min_value = input_data[gt_idx];
+    m_genotype_index = gt_idx;
+    m_allele_idx_vec_for_current_genotype = allele_idx_vec_for_current_gt_combination;
+    m_found_one_valid_value = true;
+  }
+}
+
+template<class DataType>
+void GenotypeForMinValueTracker<DataType>::determine_allele_combination_and_genotype_index_for_min_value(const std::vector<DataType>& input_data,
+    const uint64_t input_call_idx,
+    const CombineAllelesLUT& alleles_LUT,
+    const unsigned num_merged_alleles, bool NON_REF_exists,
+    const bool curr_genotype_combination_contains_missing_allele_for_input,
+    const unsigned ploidy,
+    RemappedDataWrapperBase& remapped_data,
+    std::vector<uint64_t>& num_calls_with_valid_data, DataType missing_value,
+    const std::vector<int>& remapped_allele_idx_vec_for_current_gt_combination,
+    const uint64_t remapped_gt_idx,
+    std::vector<int>& input_call_allele_idx_vec_for_current_gt_combination
+    )
+{
+  auto& min_tracker = dynamic_cast<GenotypeForMinValueTracker<DataType>&>(remapped_data);
+  min_tracker.track_minimum(input_data, input_call_allele_idx_vec_for_current_gt_combination);
+}
+
+template<class DataType>
+GenotypeForMinValueResultTuple VariantFieldHandler<DataType>::determine_allele_combination_and_genotype_index_for_min_value(
+    std::unique_ptr<VariantFieldBase>& orig_field_ptr,
+    unsigned num_merged_alleles, bool non_ref_exists, const unsigned ploidy)
+{
+  m_min_genotype_tracker.reset();
+  if(!(orig_field_ptr.get() && orig_field_ptr->is_valid()))
+    return m_min_genotype_tracker.get_tuple_for_min_value(); //returns tuple with valid flag set to false
+  //Assert that ptr is of type VariantFieldPrimitiveVectorData<DataType>
+  assert(dynamic_cast<VariantFieldPrimitiveVectorData<DataType>*>(orig_field_ptr.get()));
+  auto& orig_vector_field_data = static_cast<VariantFieldPrimitiveVectorData<DataType>*>(orig_field_ptr.get())->get();
+  m_input_call_allele_idx_vec.resize(ploidy);
+  auto num_genotypes = KnownFieldInfo::get_num_elements_given_length_descriptor(BCF_VL_G,
+      num_merged_alleles-1u, ploidy, 0u);
+  switch(ploidy)
```

Comment:
I think this switch can be split in a few functions.

Reply:
Yeah, again part of the remap restructuring :(